aten/src/ATen/native/TensorIterator.h

```diff
@@ -553,7 +550,7 @@ class CAFFE2_API TensorIteratorConfig final {
 
   c10::optional<DimVector> static_shape_ = c10::nullopt;
   c10::optional<std::pair<ScalarType, Device>> static_dtype_and_device_ = c10::nullopt;
-  bool check_mem_overlap_ = false;
+  bool check_mem_overlap_ = true;
```

Comment:
We should add some prescriptive documentation to https://github.com/pytorch/pytorch/blob/2b7108a96f3de4f20a18b44f2f64ab0e7ded3ca5/aten/src/ATen/native/TensorIterator.h#L429-L432 describing when a user should `set_check_mem_overlap = False`, and that the default is True.